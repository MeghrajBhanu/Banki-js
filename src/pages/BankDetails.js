import { useParams } from "react-router-dom";
import React, { useState, useEffect } from "react";
import { useSelector } from "react-redux";
import { get_one_id } from "../redux/actions/pan_actions";
import {useNavigate} from "react-router-dom"
import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jpg";
import bank4 from "../assets/bank4.png";
import "./BankDetails.module.css";

const image_map = new Map([
  ["Bank1", bank1],
  ["Bank2", bank2],
  ["Bank3", bank3],
  ["Bank4", bank4],
]);
const BankDetails = () => {
  let { id } = useParams();
  //const dispatch = useDispatch();
  const Navigate=useNavigate();

  const isLoading = useSelector((state) => state.pan.isLoading);
//   useEffect(() => {
//     dispatch(get_one_id(id));
//   }, []);
  const dat = useSelector((state) => state.pan.data);
  let dat1=dat.filter((item)=> item._id===id)
  let data=dat1[0]

  return (
    <div>
      {isLoading && <h1>Loading</h1>}
      {console.log(data)}
      {/*
      <div className={classes["container"]}>
        <img
          src={image_map.get(data.bankName)}
          className={classes["bank-image"]}
        />
        <section>
          <h1>{data.bankName}</h1>
        </section>
      </div>
  */}
      <div class="container" style={{ "margin-top": "60px" }}>
        <div class="card">
          <div class="container-fliud">
            <div class="wrapper row">
              <div class="preview col-md-6">
                <div class="preview-pic tab-content">
                  <div class="tab-pane active" id="pic-1">
                    <img
                      src={image_map.get(data.bankName)}
                      className="bank-image"
                    />
                  </div>
                </div>
              </div>
              <div class="details col-md-6">
                <h2 class="primary-title mt-4 mb-2">{data.bankName}</h2>

                <h4>Your details associated with {data.bankName}</h4>
                <h6>Registerd Name:{data.name}</h6>

                <p >
                 <span className="mb-4"> <strong>Email:</strong> {data.email}</span>
                  <br />
                <span className="mb-2"><strong>FixedDeposits: </strong> </span>
                  {data.FixedDeposits}
                  <br />
                  <span className="mb-2">  <strong>Total balance:</strong> {data.Balance} </span>
                  <br />
                  <span className="mb-2"><strong>AccountType:</strong> {data.AccountType} </span>
                  <br />
                  <span className="mb-2"> <strong>AccountNumber: </strong> </span>
                  {data._id}
                </p>
                <h6>Pancard: {data.pancard}</h6>
                <button onClick={()=>{
                    Navigate('/login/landing');
                }}>return</button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default BankDetails;
