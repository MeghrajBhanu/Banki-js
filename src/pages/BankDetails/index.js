import { useParams } from "react-router-dom";
import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";

import image_map from "../../utils/images";
import { flag_account } from "../../redux/actions/pan_actions";
import "./index.module.css";
/**
 * BankDetails page shows the all the details of a particular Account
 * @returns {React.FunctionComponent}
 */
const BankDetails = () => {
  let { id } = useParams();
  const dispatch = useDispatch();
  const Navigate = useNavigate();

  const isLoading = useSelector((state) => state.pan.isLoading);
  const dat = useSelector((state) => state.pan.data);
  let dat1 = dat.filter((item) => item._id === id);
  let data = dat1[0];
  const handleModal = () => {
    dispatch(flag_account(id, true));
    Navigate("/success");
  };
  return (
    <div>
      {isLoading && <h1>Loading</h1>}
      <div
        className="modal fade"
        id="exampleModal"
        tabIndex="-1"
        role="dialog"
        aria-labelledby="exampleModalLabel"
        aria-hidden="true"
      >
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title" id="exampleModalLabel">
                Attention!
              </h5>
              <button
                type="button"
                className="close"
                data-bs-dismiss="modal"
                aria-label="Close"
              >
                <span aria-hidden="true">&times;</span>
              </button>
            </div>
            <div className="modal-body">
              Are you sure you want to flag this account?
            </div>
            <div className="modal-footer">
              <button
                type="button"
                className="btn btn-secondary"
                data-bs-dismiss="modal"
              >
                No
              </button>
              <button
                type="button"
                className="btn btn-primary"
                data-bs-dismiss="modal"
                onClick={handleModal}
              >
                Yes
              </button>
            </div>
          </div>
        </div>
      </div>

      <div className="container" style={{ marginTop: "90px" }}>
        <div className="card">
          <div className="container-fliud">
            <div className="wrapper row">
              <div className="preview col-md-6">
                <img
                  src={image_map[data?.bankName]}
                  className="bank-image"
                  alt="bankLogo"
                  style={{
                    width: "100%",
                    objectFit: "cover",
                  }}
                />
              </div>
              <div className="details col-md-6">
                <h2 className="primary-title mt-4 mb-2" style={{marginLeft:"10px"}}>{data?.bankName}</h2>

                <h4 style={{marginLeft:"10px"}}>Your details associated with {data?.bankName}</h4>
                <h6 style={{marginLeft:"10px"}}>Registerd Name:{data?.name}</h6>

                <p  style={{marginLeft:"10px"}}>
                  <span className="mb-4">
                    <strong className="text-info">Email: </strong> {data?.email}
                  </span>
                  <br />
                  <span className="mb-2">
                    <strong className="text-info">FixedDeposits: </strong>
                  </span>
                  {data?.FixedDeposits}
                  <br />
                  <span className="mb-2">
                    <strong className="text-info">Total balance: </strong> {data?.Balance}
                  </span>
                  <br />
                  <span className="mb-2">
                    <strong className="text-info">AccountType: </strong> {data?.AccountType}
                  </span>
                  <br />
                  <span className="mb-2">
                    <strong className="text-info">AccountNumber: </strong>
                  </span>
                  {data?._id}
                </p>
                <h6 className="mb-5" style={{marginLeft:"10px"}}>Pancard: {data?.pancard}</h6>
                <div className="container mt-3">
                  <div className="row d-flex justify-content-space-around align-items-space-between">
                    <div className="col-lg-3 col-xl-4">
                      <button
                        className="btn btn-primary mb-2"
                        onClick={() => {
                          Navigate("/login/landing");
                        }}
                      >
                        Return
                      </button>
                    </div>
                    <div className="col-lg-3 col-xl-4">
                      <button
                        className="btn btn-primary "
                        data-bs-toggle="modal"
                        data-bs-target="#exampleModal"
                      >
                        Not your account?
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
export default BankDetails;
