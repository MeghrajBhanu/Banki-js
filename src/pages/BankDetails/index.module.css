.bank-image{
    margin:30px;
    border: 1px solid #ddd;
    border-radius: 30px;
    padding: 5px;
    object-fit:cover;
    width:100px;
}
.bank-imgage:hover {
    transform: scaleX(-1);
  }

.container{
    margin-top:200px;

}
h1{
    margin-right: 400px;
    margin-top:100px;
}
  