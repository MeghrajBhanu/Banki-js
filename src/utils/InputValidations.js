export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}