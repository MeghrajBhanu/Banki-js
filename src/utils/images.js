import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;