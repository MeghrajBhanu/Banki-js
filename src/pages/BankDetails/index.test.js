import { render, screen, waitFor } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from "react-test-renderer";
import BankDetails from ".";

describe("BankDetails", () => {
  test("snapshot test", () => {
    const tree = renderer.create(onlywrap(<BankDetails />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
