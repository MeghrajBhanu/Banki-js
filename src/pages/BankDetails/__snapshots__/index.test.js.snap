// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankDetails snapshot test 1`] = `
<div>
  <h1>
    Loading
  </h1>
  <div
    aria-hidden="true"
    aria-labelledby="exampleModalLabel"
    className="modal fade"
    id="exampleModal"
    role="dialog"
    tabIndex="-1"
  >
    <div
      className="modal-dialog"
      role="document"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <h5
            className="modal-title"
            id="exampleModalLabel"
          >
            Attention!
          </h5>
          <button
            aria-label="Close"
            className="close"
            data-bs-dismiss="modal"
            type="button"
          >
            <span
              aria-hidden="true"
            >
              Ã—
            </span>
          </button>
        </div>
        <div
          className="modal-body"
        >
          Are you sure you want to flag this account?
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-secondary"
            data-bs-dismiss="modal"
            type="button"
          >
            No
          </button>
          <button
            className="btn btn-primary"
            data-bs-dismiss="modal"
            onClick={[Function]}
            type="button"
          >
            Yes
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
    style={
      Object {
        "marginTop": "90px",
      }
    }
  >
    <div
      className="card"
    >
      <div
        className="container-fliud"
      >
        <div
          className="wrapper row"
        >
          <div
            className="preview col-md-6"
          >
            <img
              alt="bankLogo"
              className="bank-image"
              style={
                Object {
                  "objectFit": "cover",
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            className="details col-md-6"
          >
            <h2
              className="primary-title mt-4 mb-2"
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            />
            <h4
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              Your details associated with 
            </h4>
            <h6
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              Registerd Name:
            </h6>
            <p
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              <span
                className="mb-4"
              >
                <strong
                  className="text-info"
                >
                  Email: 
                </strong>
                 
              </span>
              <br />
              <span
                className="mb-2"
              >
                <strong
                  className="text-info"
                >
                  FixedDeposits: 
                </strong>
              </span>
              <br />
              <span
                className="mb-2"
              >
                <strong
                  className="text-info"
                >
                  Total balance: 
                </strong>
                 
              </span>
              <br />
              <span
                className="mb-2"
              >
                <strong
                  className="text-info"
                >
                  AccountType: 
                </strong>
                 
              </span>
              <br />
              <span
                className="mb-2"
              >
                <strong
                  className="text-info"
                >
                  AccountNumber: 
                </strong>
              </span>
            </p>
            <h6
              className="mb-5"
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              Pancard: 
            </h6>
            <div
              className="container mt-3"
            >
              <div
                className="row d-flex justify-content-space-around align-items-space-between"
              >
                <div
                  className="col-lg-3 col-xl-4"
                >
                  <button
                    className="btn btn-primary mb-2"
                    onClick={[Function]}
                  >
                    Return
                  </button>
                </div>
                <div
                  className="col-lg-3 col-xl-4"
                >
                  <button
                    className="btn btn-primary "
                    data-bs-target="#exampleModal"
                    data-bs-toggle="modal"
                  >
                    Not your account?
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
